packages/dht/src/dht/contact/SortedContactList.ts

```diff
@@ -42,7 +42,7 @@ export class SortedContactList<C extends { getPeerId: () => PeerID }> extends Co
             return
         }
         if (!this.contactsById.has(contact.getPeerId().toKey())) {
-            if (this.contactIds.length < this.maxSize) {
+            if (!this.maxSize || this.contactIds.length < this.maxSize) {
```

Comment:
```suggestion
            if ((this.maxSize === undefined) || (this.contactIds.length < this.maxSize)) {
```

Reply:
!this.maxSize allows using null "0" and "undefined" as indicator for "no limit", which I think is a good feature. I we only would allow "undefined as the indicator", we would have to throw if 0 or null was given. This would make the code more cumbersome.

Reply:
If `maxSize` would be `0`, it would pass this if statement? That would be incorrect, as we want to execute this branch only if `this.contactIds.length` is smaller than the limit.

(The value can't be `null` if we trust the type system, and we of course should trust it.)

Reply:
In general my principle is that "only compare something with undefined if there is some good reason for doing so, otherwise always test for falsiness". From my experience following this principle helps to avoid a lot of hard-to-debug errors at runtime. In this case, would you say there is some good reason for explicitly testing for uquality with undefined?

Reply:
If `maxSize` is be 0, the `addContact` should not do anything, right? I.e. it should not execute lines 46-48 in that case.  But as `!0` is `true` at line 45, it actually executes the line and adds a contact to the list.

The meaning of 0 is not always very clear if a variable has type `number | undefined`. Therefore I like to be very explicit about it when doing the comparisons:
- `if ((foo !== undefined) || (foo === 0))` tells that both `undefined` and `0` are the exceptional case
-  `if (foo !== undefined)` tells that only undefined is the exceptional case, `0` is normal valid value
- `const maxRetries = config.maxRetries ?? 123` tells that `0` is normal valid value

On the other hand these lines don't tell that information, and are buggy if `0` is actually just a normal valid value:
- `if (foo)`
- `const maxRetries = config.maxRetries || 123`